Solution

panicua opened the pull request:
Very interesting task. If I didn't miss anything in the task explanation there are no words about the rate_service function return. So it wasn't clear for me untill I found an error in tests. It should return rounded number to 1 declimal point if you return ```4.39353``` test won't accept that.

```
rate_service - method that adds a single rate to the wash station, and based on this single rate average_rating and count_of_ratings should be changed:
```

panicua committed:
Solution

oleksandra-shershen reviewed (approved):
LGTM!

YuriiKindrat commented on app/main.py:
```diff
@@ -1,8 +1,38 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int, clean_mark: int,
+                 brand: str) -> None:
```
it is better to place one/same number of args per line for better readability

YuriiKindrat commented on app/main.py:
```diff
@@ -1,8 +1,38 @@
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: int, clean_power: int,
+                 average_rating: float, count_of_ratings: int) -> None:
```
same is here

YuriiKindrat commented on app/main.py:
```diff
@@ -1,8 +1,38 @@
+        return income
+
+    def calculate_washing_price(self, car: callable) -> float:
+        calc_price = (car.comfort_class * (self.clean_power - car.clean_mark)
```
use full names for vars, f.e. washing_price, or calculated_price or full_price

YuriiKindrat commented on app/main.py:
```diff
@@ -1,8 +1,38 @@
+                self.wash_single_car(car)
+        return income
+
+    def calculate_washing_price(self, car: callable) -> float:
```
it seems to me, that car should be a Car class instance 
fix this everywhere

YuriiKindrat commented on app/main.py:
```diff
@@ -1,8 +1,38 @@
+        self.average_rating = round(average_rating, 1)
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list, income: float = 0) -> float:
```
write more descriptive type annotation for cars arg

panicua committed:
fixed flake8 and other related errors

panicua commented:
-Changed args positioning in class ```CarWashStation``` methods
-Changed ```calc_price``` variable to ```calculated_price```
-Chanded type annotation to arguments "car" inside ```CarWashStation``` class. From ```car: callable```, to ```car: Car```.
-Chanded type annotation to argument "cars" inside ```CarWashStation``` class. From ```cars: list```, to ```cars: list[Car]```

LiudmylaKulinchenko commented on app/main.py:
```diff
@@ -1,8 +1,45 @@
+
+    def serve_cars(self, cars: list[Car], income: float = 0) -> float:
+        for car in cars:
+            if car.clean_mark < self.clean_power:
```
car clean_mark should be less or equal to wash station clean_power

panicua replied:
I'll change it, but don't you think it doesn't have any logic? like why do we need to clean a car if it's ```clean_mark == clean_power```, it's gonna do nothing to cleanliness of the car. And also because of that our income not gonna change you can see the screenshot lower with calculations where ```clean_mark == clean_power```

![NVIDIA_Share_iKUiap7D8l](https://github.com/mate-academy/py-car-wash-station/assets/13889031/7c57ab64-fdaa-42c8-8d61-f7fe9a49b85c)

panicua replied:
![NVIDIA_Share_50BM8RH4Gw](https://github.com/mate-academy/py-car-wash-station/assets/13889031/c52a3b68-fd89-4b68-84ed-a8ab4bb4cee4)
![NVIDIA_Share_qE7uZ8ftG7](https://github.com/mate-academy/py-car-wash-station/assets/13889031/06406a93-ffd7-4a88-aaa0-a8bd9a2ff61f)
![NVIDIA_Share_FsumFni5oy](https://github.com/mate-academy/py-car-wash-station/assets/13889031/3e20e067-16a2-440b-8260-2ea575eb5d76)

panicua committed:
changed comparison symbol between clean_mark and clean_power

LiudmylaKulinchenko reviewed (approved):
Good job 🔥
